### PR TITLE
Enable auditlog for mothership-reconciler

### PR DESCRIPTION
**Pull request by moelsayed**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

https://github.com/kyma-incubator/reconciler/issues/291

### CODE REVIEW

File: resources/kcp/charts/mothership-reconciler/templates/deployment.yaml [hunk 1]
```diff
@@ -26,6 +26,23 @@ spec:
         - name: {{ . }}
       {{- end }}
       {{- end }}
+      {{- if .Values.global.mothership_reconciler.auditlog.persistence.enabled }}
+      initContainers:
+      - name: chwon
+        image: busybox
```
Comment (a-thaler) [inline resources/kcp/charts/mothership-reconciler/templates/deployment.yaml:32]: used image is not versioned and not coming from our registry

**Review by pPrecel — APPROVED** [review]

File: resources/kcp/charts/mothership-reconciler/templates/deployment.yaml (hunk 1 above)
Comment (kwiatekus) [reply resources/kcp/charts/mothership-reconciler/templates/deployment.yaml:32]: @moelsayed will you take care of that?
We ususally take such images from our gcr
[example](https://github.com/kyma-project/kyma/blob/main/resources/ory/charts/hydra/templates/tests/test-connection.yaml#L15)

**Review by pPrecel — APPROVED** [review]

**Review by kwiatekus — DISMISSED** [review]

File: resources/kcp/charts/mothership-reconciler/templates/deployment.yaml [hunk 2]
```diff
@@ -75,6 +103,47 @@ spec:
...
+        - sh
+        - -c
+        - |
+          # this is safe to be empty, i.e. no port specified in the URL
+          export PORT=$(echo $APP_AUDITLOG_URL | cut -d/ -f 3 | cut -d: -f2)
+          export HOST=$(echo $APP_AUDITLOG_URL | cut -d/ -f 3 | cut -d: -f1)
+          export URI=$(echo $APP_AUDITLOG_URL | cut -d/ -f4-)
+          /fluent-bit/bin/fluent-bit \
+            -e /fluent-bit/lib/flb-out_sequentialhttp.so \
+            -e /fluent-bit/lib/out_grafana_loki.so \
```
Comment (a-thaler) [inline resources/kcp/charts/mothership-reconciler/templates/deployment.yaml:123]: please remove the loki plugin, it is not used

File: resources/kcp/charts/mothership-reconciler/templates/mr-fluentbit-cm.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        # =============|==================|=================
+        Decode_Field_As   escaped_utf8    log    do_next
+        Decode_Field_As   json       log
+
+  fluent-bit.conf: |
+    [SERVICE]
+        Flush         1
+        Grace         5
+        Daemon        off
+        Log_Level     deb
```
Comment (a-thaler) [inline resources/kcp/charts/mothership-reconciler/templates/mr-fluentbit-cm.yaml:26]: Don't forget to switch to info at least  when impl is finished

File: resources/kcp/charts/mothership-reconciler/templates/mr-fluentbit-cm.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        Match            *
+        Retry_Limit      False
+        Host             ${HOST}
+        Port             ${PORT}
+        URI              ${URI}security-events
+        Header           Content-Type application/json
+        HTTP_User        ${APP_AUDITLOG_USER}
+        HTTP_Passwd      ${APP_AUDITLOG_PASSWORD}
+        Format           json_stream
+        tls              off
```
Comment (a-thaler) [inline resources/kcp/charts/mothership-reconciler/templates/mr-fluentbit-cm.yaml:63]: why tls is switched off?

File: resources/kcp/values.yaml [hunk 5]
```diff
@@ -38,6 +38,7 @@ global:
       e2e_provisioning:
         dir:
         version: "PR-930"
+    busybox: eu.gcr.io/kyma-project/external/busybox:1.32.0-v1
```
Comment (a-thaler) [inline resources/kcp/values.yaml:41]: There is a newer version 1.34.1

File: resources/kcp/charts/mothership-reconciler/templates/mr-fluentbit-cm.yaml (hunk 4 above)
Comment (moelsayed) [reply resources/kcp/charts/mothership-reconciler/templates/mr-fluentbit-cm.yaml:63]: good catch, that was for testing!

**Review by a-thaler — APPROVED** [review]

**Review by kwiatekus — APPROVED** [review]

**Review by tobiscr — DISMISSED** [review]

**Review by tobiscr — APPROVED** [review]

**Pull request merged**